File: integrations/src/main/java/framework/core/repository/ThirdPartyServiceConfigRepository.java
package framework.core.repository;

import framework.core.entity.ThirdPartyServiceConfig;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;

@Repository
public interface ThirdPartyServiceConfigRepository extends JpaRepository<ThirdPartyServiceConfig, UUID> {
    Optional<ThirdPartyServiceConfig> findByServiceAndIsActiveTrue(String serviceName);
}

File: integrations/src/main/kotlin/framework/core/repository/ThirdPartyServiceConfigRepository.kt
package framework.core.repository

import framework.core.entity.ThirdPartyServiceConfig
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface ThirdPartyServiceConfigRepository : JpaRepository<ThirdPartyServiceConfig, UUID> {
    fun findByServiceAndIsActiveTrue(serviceName: String): ThirdPartyServiceConfig?
}
